### PR TITLE
[JN-462] Survey Builder: Add new pages to a survey

### PR DESCRIPTION
**Pull request by MatthewBemis**
Adds support for adding a new blank page. Next step is being able to add questions to a page.

Screenshot 2023-07-11 at 4 39 35 PM

TO TEST:

1. Boot up Admin API and Admin UI
2. Go to the survey builder
3. Click on 'Pages' in the survey table of contents
4. Add a new page and verify that it shows up in the table of contents, pages list, and JSON editor. Note that blank pages do not show up in the Preview tab, but you can manually add a question to the JSON if you want to test that
5. Verify that you can save the survey and the new page is persisted

### CODE REVIEW

File: ui-admin/src/forms/designer/PageDesigner.tsx [hunk 1]
```diff
@@ -45,7 +47,7 @@ export const PageDesigner = (props: PageDesignerProps) => {
             setShowCreatePanelModal(true)
           }}
         >
-          Add panel
+          <FontAwesomeIcon icon={faPlus}/> Add panel
```
Comment (MatthewBemis) [inline ui-admin/src/forms/designer/PageDesigner.tsx:50]: drive-by change for consistency

File: ui-admin/src/forms/designer/PageListDesigner.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      <h2>Pages</h2>
+      <div className="mb-3">
+        <Button
+          disabled={readOnly}
+          tooltip="Add a new page."
+          variant="secondary"
+          onClick={() => {
+            onChange({
+              ...formContent,
+              pages: concat(pages, { elements: [] })
```
Comment (devonbush) [inline ui-admin/src/forms/designer/PageListDesigner.tsx]: it might be nice to have a `const newPage = () => ({ elements: []})` function, and then 
```suggestion
              pages: concat(pages, newPage())
```

File: ui-admin/src/forms/designer/PageListDesigner.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import { concat } from 'lodash/fp'
+import React from 'react'
+
+import { FormContent } from '@juniper/ui-core'
+
+import { PagesList } from './PagesList'
+import { Button } from '../../components/forms/Button'
```
Comment (devonbush) [inline ui-admin/src/forms/designer/PageListDesigner.tsx]:
```suggestion
import { Button } from 'components/forms/Button'
```

**Review by devonbush — APPROVED** [review]
looks good

File: ui-admin/src/forms/designer/PageListDesigner.tsx (hunk 2 above)
Comment (MatthewBemis) [reply ui-admin/src/forms/designer/PageListDesigner.tsx]: done in 9ed5cd35

**Comment by breilly2** [comment]
Initial drive-by observation: reordering pages is clunky. The up/down buttons do work, but the pages are identified by their number, which changes when it moves, so it looks like clicking them does nothing unless you're watching the sidebar.

**Review by breilly2 — APPROVED** [review]

**Comment by MatthewBemis** [comment]
@breilly2 I completely agree, that confused me as well. We're very much still in the bare-bones phase of this. My goal with this current batch of PRs is to unlock basic functionality so Erin and Jeff can start playing around with start-to-finish survey creation and get a clearer idea of how certain interactions should work. We're definitely going to iterate on this stuff a lot once this gets into their hands.
